views/_rtm_rest_api_client.njk

```diff
@@ -0,0 +1,217 @@
+
+### Query the history messages
+
+This interface requires master key.
+
+Invoke this interface can query messages sent by specified client_id in One-on-One Chatting/Group Chats and Chats Room.
+```sh
+curl -X GET \
+  -H "X-LC-Id: {{appid}}" \
+  -H "X-LC-Key: {{masterkey}},master" \
+  https://{{host}}/1.2/rtm/clients/{client_id}/messages
+```
+
+parameters and the returned value refer to [One-on-One Chatting/Group Chats query the history messages](realtime_rest_api.html#Query history messages) interface.
+
+### Add Blocked List 
+
+This interface requires master key.
+
+One client is permitted to add one group chat/ chatroom / Official Account to the Blocked list. Then no other clients is prohibited from joining in the conversations.
+Currently client cannot add another client into the Blocked List.
+
+```sh
+curl -X POST \
+  -H "X-LC-Id: {{appid}}" \
+  -H "X-LC-Key: {{masterkey}},master" \
+  -H "Content-Type: application/json" \
+  -d '{"conv_id": ""}' \
+  https://{{host}}/1.2/rtm/clients/{client_id}/blacklists
+```
+
+Parameters | Optionality | Description
+---|---|---
+conv_id | required | blocked Group chats/chatroom/Official Account ID
+
+return
+```
+{}
+```
+
+### Remove Blocked List
+
+```sh
+curl -X DELETE \
+  -H "X-LC-Id: {{appid}}" \
+  -H "X-LC-Key: {{masterkey}},master" \
+  -H "Content-Type: application/json" \
+  -d '{"conv_id": ""}' \
+  https://{{host}}/1.2/rtm/clients/{client_id}/blacklists
+```
+
+return
+```
+{}
+```
+
+### Query Blocked List 
+
+```sh
+curl -X GET \
+  -H "X-LC-Id: {{appid}}" \
+  -H "X-LC-Key: {{masterkey}},master" \
+  https://{{host}}/1.2/rtm/clients/{client_id}/blacklists
+```
+
+Parameters | Optionality | Description
+---|---|---
+limit | Optional | together with next to realize pagination , default on 10
+next | Optional | returned on the first query, the following query with this parameter to realize pagination
+
+return
+
+```
+{"conv_ids"=>["conv1"], "next"=>"1"}
+```
+
+### Retrieve sign-in signature
+
+This interface allows app using [AV.User](rest_api.html#User-1) expressly realize sign-in authentication. Sign-in authentication is closed by default, go to [Dashboard > Messaging > LeanMessage > Settings > LeanMessage settings](/dashboard/messaging.html?appid={{appid}}#/message/realtime/conf) , tick **Verify signatures for logging in** to enable it.
+
+```sh
+curl -X GET \
+  -H "X-LC-Id: {{appid}}" \
+  -H "X-LC-Key: {{appkey}}" \
+  -G \
+  --data-urlencode 'session_token=some-token' \
+  https://{{host}}/1.2/rtm/clients/sign
+```
+
+Parameters | Optionality | Description
+---|---|---
+session_token | required | sessionToken for AV.User
+
+return
+
+```
+{"nonce": "", "timestamp": "", "client_id": "", "signature": ""}
+```
+
+To facilitate the Fine-grained control for users, realizing functions self-customized (Blocked List), this interface provides hook function `_rtmClientSign`. After Verifying the sessionToken to invoke, the input parameter is JSON object comprised of AV.User.
```

Comment:
Fine -> fine

provides hook function -> provides a hook function

After Verifying the sessionToken to invoke, -> This hook will be invoked after verifying the sessionToken.

the input parameter is JSON object comprised of AV.User. -> Its parameter is a JSON object of `AV.User`.